homeassistant/components/permobil/sensor.py

```diff
@@ -0,0 +1,361 @@
+"""Platform for sensor integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from mypermobil import (
+    BATTERY_AMPERE_HOURS_LEFT,
+    BATTERY_CHARGE_TIME_LEFT,
+    BATTERY_CHARGING,
+    BATTERY_DISTANCE_LEFT,
+    BATTERY_DISTANCE_UNIT,
+    BATTERY_INDOOR_DRIVE_TIME,
+    BATTERY_MAX_AMPERE_HOURS,
+    BATTERY_MAX_DISTANCE_LEFT,
+    BATTERY_STATE_OF_CHARGE,
+    BATTERY_STATE_OF_HEALTH,
+    RECORDS_DISTANCE,
+    RECORDS_SEATING,
+    USAGE_ADJUSTMENTS,
+    USAGE_DISTANCE,
+    MyPermobil,
+    MyPermobilAPIException,
+    MyPermobilClientException,
+)
+
+from homeassistant import config_entries
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+)
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorStateClass,
+)
+from homeassistant.const import (
+    CONF_CODE,
+    CONF_EMAIL,
+    CONF_REGION,
+    CONF_TOKEN,
+    CONF_TTL,
+    PERCENTAGE,
+    UnitOfEnergy,
+    UnitOfLength,
+    UnitOfTime,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from .const import APPLICATION, BATTERY_ASSUMED_VOLTAGE, DOMAIN, KM, MILES
+
+
+def setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the sensor platform."""
+
+
+_LOGGER = logging.getLogger(__name__)
+SCAN_INTERVAL = timedelta(seconds=50)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Create sensors from a config entry created in the integrations UI."""
+    config = hass.data[DOMAIN][config_entry.entry_id]
+    if config_entry.options:
+        config.update(config_entry.options)
+    session = hass.helpers.aiohttp_client.async_get_clientsession()
+    # create the API object from the config
+    p_api = MyPermobil(
+        application=APPLICATION,
+        session=session,
+        email=config.get(CONF_EMAIL),
+        region=config.get(CONF_REGION),
+        code=config.get(CONF_CODE),
+        token=config.get(CONF_TOKEN),
+        expiration_date=config.get(CONF_TTL),
+    )
+
+    p_api.self_authenticate()
+
+    p_api_unit: str = KM
+    try:
+        # find out what unit of distance the user has set
+        p_api_unit = str(await p_api.request_item(BATTERY_DISTANCE_UNIT))
+    except (MyPermobilClientException, MyPermobilAPIException) as err:
+        _LOGGER.error("Permobil: %s", err)
```

Comment:
What would be the consequence of this error?

Reply:
This is not needed anymore since I can default the API to always return distances in KM and to my understanding HA will convert the units if necessary.